sharktank/sharktank/ops/sharded_impls.py

```diff
@@ -41,7 +41,7 @@ def all_gather_split(
     shards = [
         cat(
             [
-                transfer_to_logical_device(shard, i) if j != i else shard
+                transfer_to_logical_device(shard, i)
```

Comment:
Why did you remove this optimization?

Reply:
It causes compilation issues with the model sadly. It happens during some fusion passes so its likely a compiler bug but we can reintroduce it later on.